File: CUSTOR.EICOnline.Angular/src/app/model/address.ts
export class Region {
  public RegionId: string;
  public Description: string;
  public DescriptionEnglish: string;

  constructor(RegionId?: string, Description?: string, DescriptionEnglish?: string) {
    this.RegionId = RegionId;
    this.Description = Description;
    this.DescriptionEnglish = DescriptionEnglish;
  }
}

export class Zone {
  public ZoneId: string;
  public RegionId: string;
  public Description: string;
  public DescriptionEnglish: string;
  public Region: Region;

  constructor(ZoneId?: string, RegionId?: string, Description?: string, DescriptionEnglish?: string, RegionName?: Region) {
    this.ZoneId = ZoneId;
    this.RegionId = RegionId;
    this.Description = Description;
    this.DescriptionEnglish = DescriptionEnglish;
    this.Region = RegionName;
  }
}

export class Woreda {
  public RegionId: string;
  public ZoneId: string;
  public WoredaId: string;
  public Description: string;
  public DescriptionEnglish: string;
  public Region: Region;
  public Zone: Zone;

  constructor(RegionId?: string, ZoneId?: string, WoredaId?: string, Description?: string, DescriptionEnglish?: string, RegionName?: Region, ZoneName?: Zone) {
    this.RegionId = RegionId;
    this.ZoneId = ZoneId;
    this.WoredaId = WoredaId;
    this.Description = Description;
    this.DescriptionEnglish = DescriptionEnglish;
    this.Region = RegionName;
    this.Zone = ZoneName;
  }
}

export class Kebele {
  public RegionId: string;
  public ZoneId: string;
  public KebeleId: string;
  public WoredaId: string;
  public Description: string;
  public DescriptionEnglish: string;
  public Region: Region;
  public Zone: Zone;
  public Woreda: Woreda;

  constructor(RegionId?: string, ZoneId?: string, KebeleId?: string, WoredaId?: string, Description?: string, DescriptionEnglish?: string, RegionName?: Region, ZoneName?: Zone, WoredaName?: Woreda) {
    this.RegionId = RegionId;
    this.ZoneId = ZoneId;
    this.KebeleId = KebeleId;
    this.WoredaId = WoredaId;
    this.Description = Description;
    this.DescriptionEnglish = DescriptionEnglish;
    this.Region = RegionName;
    this.Zone = ZoneName;
    this.Woreda = WoredaName;
  }
}

File: CUSTOR.EICOnline.DAL/EntityLayer/Address.cs
using System;
using System.Collections.Generic;

namespace CUSTOR.EICOnline.DAL.EntityLayer
{
  public partial class Address
  {
    public Address()
    {
      Associate = new HashSet<Associate>();
      //Investor = new HashSet<Investor>();
    }

    public int AddressId { get; set; }
    public int ParentId { get; set; }
    public int AddressType { get; set; }
    public bool IsMainOffice { get; set; }
    public string SpecificAreaName { get; set; }
    public string RegionId { get; set; }
    public string TownId { get; set; }
    public string ZoneId { get; set; }
    public string WoredaId { get; set; }
    public string KebeleId { get; set; }
    public string HouseNo { get; set; }
    public string TeleNo { get; set; }
    public string Pobox { get; set; }
    public string Fax { get; set; }
    public string CellPhoneNo { get; set; }
    public string Email { get; set; }
    public string OtherAddress { get; set; }

    public bool? IsIndustrialPark { get; set; }
    public int? IndustrialParkId { get; set; }


    public string Remark { get; set; }
    public bool? IsActive { get; set; }
    public bool IsDeleted { get; set; }
    public DateTime EventDatetime { get; set; }
    public int CreatedUserId { get; set; }
    public string CreatedUserName { get; set; }
    public DateTime? UpdatedEventDatetime { get; set; }
    public int? UpdatedUserId { get; set; }
    public string UpdatedUserName { get; set; }
    public Guid? ObjectId { get; set; }

    public Kebele Kebele { get; set; }
    public Region Region { get; set; }
    public Town Town { get; set; }
    public Woreda Woreda { get; set; }
    public Zone Zone { get; set; }
    public ICollection<Associate> Associate { get; set; }
    //public ICollection<Investor> Investor { get; set; }
  }
}